nixos/modules/services/networking/wireguard.nix

```diff
@@ -0,0 +1,165 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.networking.wireguard;
+
+  kernel = config.boot.kernelPackages;
+
+  # interface options
+
+  interfaceOpts = { name, ... }: {
+
+    options = {
+
+      ip = mkOption {
+        example = "192.168.2.1";
+        type = types.str;
+        description = "The ip address added to the interface.";
```

Comment:
> Do you think that this option should allow multiple IPs? 
Yes, probably. Some people like to have a v4 and a v6 address. This part, which amounts to calling `ip addr add`, is the same as with setting up any network interface. What does the NixOS configuration do for ordinary Ethernet cards? If it allows adding multiple IPs to `eth0`, then this here should allow adding multiple IPs to `wg0`. In other words, this option is the most ordinary and standard one of the module.